### PR TITLE
DataSources: Add datasource fetching + querying interface

### PR DESCRIPTION
**Pull request by wbrowne**
**What is this feature?**

Adds a `Querier` interface so that datasources can follow the same general shape.

**Special notes for your reviewer:**

Follow up to come with separating standalone into different wiresets https://github.com/grafana/grafana/tree/dsAPIServerInterfacesWithWire

Also - we could technically have a different impl for non standalone in enterprise too (see here for an example of that https://github.com/grafana/grafana-enterprise/pull/6181)

Please check that:
- [X] It works as expected from a user's perspective.
- [X] If this is a pre-GA feature, it is behind a feature toggle.
- [X] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by andresmgot — APPROVED** [review]
FWIW, this LGTM it sets another good step towards a possible MVP :+1:

**Review by marefr — APPROVED** [review]
Agree LGTM with a nit :)

File: pkg/services/pluginsintegration/plugincontext/plugincontext.go [hunk 1]
```diff
@@ -128,6 +131,48 @@ func (p *Provider) GetWithDataSource(ctx context.Context, pluginID string, user
 	return pCtx, nil
 }
 
+func (p *Provider) PluginContextForDataSource(ctx context.Context, pluginID, name string) (backend.PluginContext, error) {
```
Comment (marefr) [inline pkg/services/pluginsintegration/plugincontext/plugincontext.go:134]: It's interesting that the comment for `Get` above says "If datasourceUID is not empty string
// then PluginContext.DataSourceInstanceSettings will be resolved and appended to
// returned context." which is not true :smile: Feel free to remove if you want.

**Review by marefr — APPROVED** [review]
LGTM with a comment, but can skip that

File: pkg/services/pluginsintegration/plugincontext/plugincontext.go (hunk 1 above)
Comment (wbrowne) [reply pkg/services/pluginsintegration/plugincontext/plugincontext.go:134]: Added to a follow up (even if this stuff is short lived) https://github.com/grafana/grafana/pull/80906

**Pull request merged**